api/handlers/search.js

```diff
@@ -0,0 +1,157 @@
+const aws4 = require('aws4');
+const Pool = require('pg').Pool;
+const https = require('https');
+require('dotenv').config();
+
+const pool = new Pool({
+    user: 'masteruser',
+    password: 'ecommdbpw',
+    host: 'ecommdb.covnxnivrrc5.ap-southeast-2.rds.amazonaws.com',
+    database: 'ecommdb',
+    port: 5432,
+    ssl: true
+});
+
+module.exports.bulkIndexProductData = async (event) => {
+    try {
+        const productData = await pool.query('SELECT * FROM products');
+        const products = productData.rows;
+        const bulkBody = [];
+
+        // loop through products and create index reqs
+        products.forEach((product) => {
+            bulkBody.push(
+                { index: { _index: 'product_index', _id: product.id } },
+                {
+                    product_name: product.product_name,
+                    product_description: product.product_description,
+                    category: product.category,
+                    price: product.price,
+                    available_quantity: product.available_quantity,
+                    image_url: product.image_url
+                }
+            );
+        });
+
+        const requestBody = bulkBody.map(JSON.stringify).join('\n') + '\n';
+
+        const requestOptions = aws4.sign({
+            hostname: 'search-product-search2-gpf6dh65zyaj3f2eg5i3tfzfwa.ap-southeast-2.es.amazonaws.com',
+            service: 'es',
+            region: 'ap-southeast-2',
+            method: 'POST',
+            path: '/product_index/_bulk',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: requestBody
+        },  {
+          accessKeyId: process.env.aws_access_key,
+          secretAccessKey: process.env.aws_secret_key,
+        });
+
+        // Make the HTTP request
+        const response = await new Promise((resolve, reject) => {
+            const req = https.request(requestOptions, (res) => {
+                let data = '';
+                res.on('data', (chunk) => {
+                    data += chunk;
+                });
+                res.on('end', () => {
+                    resolve(data);
+                });
+            });
+
+            req.on('error', (error) => {
+                reject(error);
+            });
+
+            req.write(requestOptions.body);
+            req.end();
+        });
+
+        console.log(response);
```

Comment:
This is OK during development but putting a whole response into the log in production is generally a bad idea. If you really do want to log it then include a message along with the response so that you know what it is when you're looking at the logs.